docs/fuse-box/tutorials/mobile-swap-tokens-guide.mdx

```diff
@@ -0,0 +1,404 @@
+
+### **Perform Token Swap**
+   ```dart
+     try {
+       final res = await fuseSDK.swapTokens(
+         TradeRequest(
+           inputToken: nativeTokenAddress,
+           outputToken: usdcTokenAddress,
+           inputAmount: BigInt.parse('100000000000000000'),
+           exactIn: true,
+         ),
+       );
+
+       setState(() {
+         _swapStatus = 'UserOpHash: ${res.userOpHash}';
+       });
+
+       final ev = await res.wait();
+       setState(() {
+         _swapStatus = 'Transaction hash: ${ev?.transactionHash}';
+       });
+
+       print('UserOpHash: ${res.userOpHash}');
+     } catch (error) {
+       setState(() {
+         _swapStatus = 'Error: $error';
+       });
+     } finally {
+       setState(() {
+         _isLoading = false;
+       });
+     }
+   }
+   ```
+   
+We perform the token swap operation inside the `try` block using `fuseSDK.swapTokens()`. We set the `_swapStatus` based on the result of the operation. The result contains `userOpHash`, which is printed to the console. If an error occurs during the operation, we update `_swapStatus` with the error message. Finally, we set `_isLoading` to `false` to indicate that the operation is complete.
+
+###  **UI Build Method**
+   ```dart
+     @override
+     Widget build(BuildContext context) {
+       return Scaffold(
+         appBar: AppBar(
+           title: Text('Token Swap'),
+         ),
+         body: Center(
+           child: Column(
+             mainAxisAlignment: MainAxisAlignment.center,
+             children: [
+               _isLoading
+                   ? CircularProgressIndicator()
+                   : ElevatedButton(
+                       onPressed: _swapTokens,
+                       child: Text('Swap Tokens'),
+                     ),
+               SizedBox(height: 20),
+               Text(_swapStatus),
+             ],
+           ),
+         ),
+       );
+     }
+   }
+   ```
+  
+In the `build` method, we return a `Scaffold` widget, which provides the basic structure for the screen. The `AppBar` contains the title 'Token Swap'. The `body` consists of a `Center` widget containing a `Column` with a `CircularProgressIndicator` or an `ElevatedButton` based on the `_isLoading` state. Below the button, we display the `_swapStatus` using a `Text` widget.
+
+Save the file. Run the application using the command
+
+```bash
+flutter run
+```
+
+At this point, you can confirm that the SWAP functionality works! The next step is a bonus step to improve the User Experience, and we will update the code to include a Dropdown for selecting currencies and an input field to specify a particular amount. Add the Token address variables:
+
+```dart
+ final amount = BigInt.parse(_amountController.text.trim());
```

Comment:
You using here undeclared controller (i can see it in source code, but its didnt mentioned in this tutorial)

Reply:
Awesome. Thank you for looking through the code. There is all the code. If you are referring to the code separated into parts, it is so that there is no repetition of code already explained in other sections of the tutorial.